atomic_reactor/plugins/pre_distribution_scope.py

```diff
@@ -0,0 +1,107 @@
+"""
+Copyright (c) 2016 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, unicode_literals
+
+from atomic_reactor.plugin import PreBuildPlugin
+from dockerfile_parse import DockerfileParser
+
+
+class NothingToCheck(Exception):
+    pass
+
+
+class DisallowedDistributionScope(Exception):
+    pass
+
+
+class DistributionScopePlugin(PreBuildPlugin):
+    """Apply distribution-scope rule.
+
+    Possible values for this label are:
+
+    (most restrictive)
+    - private
+    - authoritative-source-only
+    - restricted
+    - public
+    (least restrictive)
+
+    The rule we want to apply is to prevent images from having a
+    less restrictive scope than their parent images.
+
+    If the distribution-scope for this image is set to a less
+    restrictive value than the distribution-scope label inherited from
+    the parent image, fail the build.
+    """
+
+    # Name of this plugin
+    key = 'distribution_scope'
+
+    # The label and its possible values.
+    # https://github.com/projectatomic/ContainerApplicationGenericLabels
+
+    # LABEL name used for specifying scope
+    SCOPE_LABEL = 'distribution-scope'
+
+    # Valid distribution-scope choice values, most restricted first
+    SCOPE_NAME = [
+        "private",
+        "authoritative-source-only",
+        "restricted",
+        "public",
+    ]
+
+    def __init__(self, tasker, workflow):
+        """
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        """
+
+        self.workflow = workflow
+
+    def get_scope(self, which, labels):
+        """
+        :param which: str, description of the image this belongs to
+        :param labels: dict, labels on the image
+        """
+
+        try:
+            scope_choice = labels[self.SCOPE_LABEL]
+        except KeyError:
+            self.log.debug("no distribution scope set for %s image", which)
```

Comment:
That's fine for the parent image, but we probably want the default value for the current image to be 'inherit from parent', which is technically what will happen once we build it.

I figured it was easiest to explicitly not perform checks when that label is not present.